### ISSUE TITLE
Issues trying to install plaso 1.3 on Mac OS X El Capitan

### ISSUE THREAD

**Issue by Lou511** [log2timeline/plaso#404]
While running Plaso installer for Mac, some packages successfully installed, and some failed.  I tried to do a manual installation of one of the packages - "dfvfs-20150708" - to see if any error messages would return.  I received an alert indicating the package was "not compatible" with my version of OS X.  I am running Yosemite.  I have included the output from the installation process (conducted within Terminal).

Thank you in advance for any help.
```
===============================================================
        PLASO INSTALLER
===============================================================
This script requires root privileges. Running: sudo.
Password:
Installing packages.
find: /Volumes/plaso-1.3.0/.Trashes: Permission denied
installer: Package name is artifacts-20150409
installer: Upgrading at base path /
installer: The upgrade failed (The Installer encountered an error that caused the installation to fail. Contact the software manufacturer for assistance.)
installer: Package name is bencode-1.0
installer: Upgrading at base path /
installer: The upgrade was successful.
installer: Package name is binplist-0.1.5
installer: Upgrading at base path /
installer: The upgrade was successful.
installer: Package name is construct-2.5.2
installer: Upgrading at base path /
installer: The upgrade was successful.
installer: Package name is dateutil-2.4.2
installer: Upgrading at base path /
installer: The upgrade was successful.
installer: Package name is dfvfs-20150708
installer: Upgrading at base path /
installer: The upgrade failed (The Installer encountered an error that caused the installation to fail. Contact the software manufacturer for assistance.)
installer: Package name is docopt-0.6.2
installer: Upgrading at base path /
installer: The upgrade was successful.
installer: Package name is dpkt-1.8.6.2
installer: Upgrading at base path /
installer: The upgrade was successful.
installer: Package name is Flask-0.10.1
installer: Upgrading at base path /
installer: The upgrade was successful.
installer: Package name is Flask-Sockets-0.1
installer: Upgrading at base path /
installer: The upgrade was successful.
installer: Package name is hachoir-core-1.3.3
installer: Upgrading at base path /
installer: The upgrade was successful.
installer: Package name is hachoir-metadata-1.3.3
installer: Upgrading at base path /
installer: The upgrade was successful.
installer: Package name is hachoir-parser-1.3.4
installer: Upgrading at base path /
installer: The upgrade was successful.
installer: Package name is ipython.1.2.1
installer: Upgrading at base path /
installer: The upgrade failed (The Installer encountered an error that caused the installation to fail. Contact the software manufacturer for assistance.)
installer: Package name is libbde-20150204
installer: Upgrading at base path /
installer: The upgrade failed (The Installer encountered an error that caused the installation to fail. Contact the software manufacturer for assistance.)
installer: Package name is libcaes-20150514
installer: Upgrading at base path /
installer: The upgrade failed (The Installer encountered an error that caused the installation to fail. Contact the software manufacturer for assistance.)
installer: Package name is libesedb-20150409
installer: Upgrading at base path /
installer: The upgrade failed (The Installer encountered an error that caused the installation to fail. Contact the software manufacturer for assistance.)
installer: Package name is libevt-20150706
installer: Upgrading at base path /
installer: The upgrade failed (The Installer encountered an error that caused the installation to fail. Contact the software manufacturer for assistance.)
installer: Package name is libevtx-20150105
installer: Upgrading at base path /
installer: The upgrade failed (The Installer encountered an error that caused the installation to fail. Contact the software manufacturer for assistance.)
installer: Package name is libewf-20140608
installer: Upgrading at base path /
installer: The upgrade failed (The Installer encountered an error that caused the installation to fail. Contact the software manufacturer for assistance.)
installer: Package name is libfwsi-20150701
installer: Upgrading at base path /
installer: The upgrade failed (The Installer encountered an error that caused the installation to fail. Contact the software manufacturer for assistance.)
installer: Package name is liblnk-20150617
installer: Upgrading at base path /
installer: The upgrade failed (The Installer encountered an error that caused the installation to fail. Contact the software manufacturer for assistance.)
installer: Package name is libmsiecf-20150708
installer: Upgrading at base path /
installer: The upgrade failed (The Installer encountered an error that caused the installation to fail. Contact the software manufacturer for assistance.)
installer: Package name is libolecf-20150630
installer: Upgrading at base path /
installer: The upgrade failed (The Installer encountered an error that caused the installation to fail. Contact the software manufacturer for assistance.)
installer: Package name is libqcow-20150105
installer: Upgrading at base path /
installer: The upgrade failed (The Installer encountered an error that caused the installation to fail. Contact the software manufacturer for assistance.)
installer: Package name is libregf-20150704
installer: Upgrading at base path /
installer: The upgrade failed (The Installer encountered an error that caused the installation to fail. Contact the software manufacturer for assistance.)
installer: Package name is libsigscan-20150714
installer: Upgrading at base path /
installer: The upgrade failed (The Installer encountered an error that caused the installation to fail. Contact the software manufacturer for assistance.)
installer: Package name is libsmdev-20150105
installer: Upgrading at base path /
installer: The upgrade failed (The Installer encountered an error that caused the installation to fail. Contact the software manufacturer for assistance.)
installer: Package name is libsmraw-20150105
installer: Upgrading at base path /
installer: The upgrade failed (The Installer encountered an error that caused the installation to fail. Contact the software manufacturer for assistance.)
installer: Package name is libvhdi-20150110
installer: Upgrading at base path /
installer: The upgrade failed (The Installer encountered an error that caused the installation to fail. Contact the software manufacturer for assistance.)
installer: Package name is libvmdk-20150516
installer: Upgrading at base path /
installer: The upgrade failed (The Installer encountered an error that caused the installation to fail. Contact the software manufacturer for assistance.)
installer: Package name is libvshadow-20150106
installer: Upgrading at base path /
installer: The upgrade failed (The Installer encountered an error that caused the installation to fail. Contact the software manufacturer for assistance.)
installer: Package name is mock-1.0.1
installer: Upgrading at base path /
installer: The upgrade was successful.
installer: Package name is pefile-1.2.10-139
installer: Upgrading at base path /
installer: The upgrade was successful.
installer: Package name is protobuf-2.6.1
installer: Upgrading at base path /
installer: The upgrade failed (The Installer encountered an error that caused the installation to fail. Contact the software manufacturer for assistance.)
installer: Package name is psutil-1.2.1
installer: Upgrading at base path /
installer: The upgrade failed (The Installer encountered an error that caused the installation to fail. Contact the software manufacturer for assistance.)
installer: Package name is pyparsing-2.0.3
installer: Upgrading at base path /
installer: The upgrade was successful.
installer: Package name is pysqlite-2.6.3
installer: Upgrading at base path /
installer: The upgrade failed (The Installer encountered an error that caused the installation to fail. Contact the software manufacturer for assistance.)
installer: Package name is python-gflags-2.0
installer: Upgrading at base path /
installer: The upgrade failed (The Installer encountered an error that caused the installation to fail. Contact the software manufacturer for assistance.)
installer: Package name is python-plaso-1.3.0
installer: Upgrading at base path /
installer: The upgrade failed (The Installer encountered an error that caused the installation to fail. Contact the software manufacturer for assistance.)
installer: Package name is python-protobuf-2.6.1
installer: Upgrading at base path /
installer: The upgrade was successful.
installer: Package name is pytsk3-4.1.3_20140506
installer: Upgrading at base path /
installer: The upgrade failed (The Installer encountered an error that caused the installation to fail. Contact the software manufacturer for assistance.)
installer: Package name is pytz-2015.4
installer: Upgrading at base path /
installer: The upgrade was successful.
installer: Package name is PyYAML-3.11
installer: Upgrading at base path /
installer: The upgrade was successful.
installer: Package name is readline-6.2.4.1
installer: Upgrading at base path /
installer: The upgrade was successful.
installer: Package name is requests-2.7.0
installer: Upgrading at base path /
installer: The upgrade was successful.
installer: Package name is six-1.9.0
installer: Upgrading at base path /
installer: The upgrade was successful.
installer: Package name is sleuthkit-4.1.3
installer: Upgrading at base path /
installer: The upgrade failed (The Installer encountered an error that caused the installation to fail. Contact the software manufacturer for assistance.)
installer: Package name is talloc-2.1.1
installer: Upgrading at base path /
installer: The upgrade failed (The Installer encountered an error that caused the installation to fail. Contact the software manufacturer for assistance.)
installer: Package name is XlsxWriter-0.7.3
installer: Upgrading at base path /
installer: The upgrade was successful.
Done.
Two versions of the pyparsing module exist. Deleting the system provided one.
```
```
Luiss-MBP:plaso-1.3.0 luisemartinez$ log2timeline.py -h
Traceback (most recent call last):
  File "/usr/local/bin/log2timeline.py", line 18, in <module>
    from dfvfs.helpers import source_scanner
  File "/Library/Python/2.7/site-packages/dfvfs/helpers/source_scanner.py", line 20, in <module>
    from dfvfs.analyzer import analyzer
  File "/Library/Python/2.7/site-packages/dfvfs/analyzer/__init__.py", line 3, in <module>
    from dfvfs.analyzer import bde_analyzer_helper
  File "/Library/Python/2.7/site-packages/dfvfs/analyzer/bde_analyzer_helper.py", line 4, in <module>
    from dfvfs.analyzer import analyzer
  File "/Library/Python/2.7/site-packages/dfvfs/analyzer/analyzer.py", line 4, in <module>
    import pysigscan
ImportError: dlopen(/Library/Python/2.7/site-packages/pysigscan.so, 2): Library not loaded: /usr/lib/libsigscan.1.dylib
  Referenced from: /Library/Python/2.7/site-packages/pysigscan.so
  Reason: image not found
```
```
Luiss-MBP:plaso-1.3.0 luisemartinez$ log2timeline.py -v
Traceback (most recent call last):
  File "/usr/local/bin/log2timeline.py", line 18, in <module>
    from dfvfs.helpers import source_scanner
  File "/Library/Python/2.7/site-packages/dfvfs/helpers/source_scanner.py", line 20, in <module>
    from dfvfs.analyzer import analyzer
  File "/Library/Python/2.7/site-packages/dfvfs/analyzer/__init__.py", line 3, in <module>
    from dfvfs.analyzer import bde_analyzer_helper
  File "/Library/Python/2.7/site-packages/dfvfs/analyzer/bde_analyzer_helper.py", line 4, in <module>
    from dfvfs.analyzer import analyzer
  File "/Library/Python/2.7/site-packages/dfvfs/analyzer/analyzer.py", line 4, in <module>
    import pysigscan
ImportError: dlopen(/Library/Python/2.7/site-packages/pysigscan.so, 2): Library not loaded: /usr/lib/libsigscan.1.dylib
  Referenced from: /Library/Python/2.7/site-packages/pysigscan.so
  Reason: image not found
Luiss-MBP:plaso-1.3.0 luisemartinez$ 
```

**Comment by joachimmetz** [comment]
>  I received an alert indicating the package was "not compatible" with my version of OS X.

Where is this error in your output? Otherwise how did you come to this?

From the output it looks like "/usr/lib/libsigscan.1.dylib" is missing.

* can you check if this file exists
* can you check if reinstalling pysigscan might solve that issue

**Comment by Lou511** [comment]
Here is the error I received.  I encountered after I navigated inside of plaso-1.3.0 to the packages folder, and selected the dfvfs package.  I selected this package, because it was the first package that did not install successfully (as you can see in the logs above).

libsigscan.1.dylib is missing, but doesn't its functionality depend on a successful installation of plaso?  Or is that library something totally independent of plaso?  Thank you for your help.


![dfvfs manual install copy](https://cloud.githubusercontent.com/assets/13272787/11014989/2051ba20-851a-11e5-972e-53a57b8922ed.jpg)

**Comment by joachimmetz** [comment]
If you are running El Capitan the "warning" makes sense because the packages have been built for Yosemite. If you're running El Capitan, Apple changing the packaging could be the root cause, also see: https://georgegarside.com/blog/osx/package-incompatible-installer/

However the "warning" lacks necessary details to assess what is "incompatible".

> libsigscan.1.dylib is missing, but doesn't its functionality depend on a successful installation of plaso? 

Not sure what you mean. You'll need libsigscan to run plaso. It is a separate project to allow reuse.

**Comment by grgarside** [comment]
I'm the guy that wrote the blog post that @joachimmetz linked to. I've had a look at the package and the reason that the warning appears is because the package attempts to write some text files (specifically Acknowledgements, Authors, License and Readme) to the following directory:

    /System/Library/Frameworks/Python.framework/Versions/2.7/share/doc/dfvfs/

System Integrity Protection protects Python.framework from modification by packages not signed by Apple, thus Installer is unable to write these files. The warning appears because Installer detects that the package may be trying to do something like this.

If you wish to write these files, you need to disable SIP—follow the instructions in the aforementioned link. You can re-enable SIP after running the installer.

**Comment by joachimmetz** [comment]
@grgarside thanks for the more detailed explanation

To my knowledge "System Integrity Protection" was added in El Capitan and we only provide OSX packages for Yosemite at the moment (https://github.com/log2timeline/l2tbinaries). I'll have a more detailed look at this for the next release and see what the most appropriate solution is.

**Comment by Lou511** [comment]
Ok.  Thank you for the follow up.  I will disable SIP and attempt to
reinstall.

Luis

**Comment by joachimmetz** [comment]
Issue will be addressed in plaso 1.4 release. Closing this issue.
